### PR TITLE
Initial PR for pfSense-pkg-WireGuard

### PR DESCRIPTION
**Pull request by theonemcdonald**
This is the first PR for the pfSense-pkg-WireGuard package.

https://github.com/theonemcdonald/pfSense-pkg-WireGuard

https://redmine.pfsense.org/issues/11752

### CODE REVIEW

File: net/pfSense-pkg-WireGuard/Makefile [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# $FreeBSD$
+
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/Makefile:2]: These tags were removed all over FreeBSD ports tree since it moved from svn to git.  Please remove them

File: net/pfSense-pkg-WireGuard/Makefile [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+PORTNAME=	pfSense-pkg-WireGuard
+PORTVERSION=	0.0.6
+PORREVISION=	3
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/Makefile:5]: It's the first version added to the project, PORTREVISION is not needed

File: net/pfSense-pkg-WireGuard/Makefile [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+CATEGORIES=	net
+MASTER_SITES=	# empty
+DISTFILES=	# empty
+EXTRACT_ONLY=	# empty
+
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/Makefile:10]: Remove extra blank line

File: net/pfSense-pkg-WireGuard/Makefile [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+MAINTAINER=	rcmcdonald91@gmail.com
+COMMENT=	pfSense package WireGuard
+
+LICENSE=	APACHE20
+
+RUN_DEPENDS=	${LOCALBASE}/bin/bash:shells/bash \
+		wg:net/wireguard-tools \
+		wg-quick:net/wireguard-tools \
+		${KMODDIR}/if_wg.ko:net/wireguard-kmod
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/Makefile:20]: Use path to detect a port is the old way, today use `portname>=0:category/port-directory` is preferred, as follow:

```
RUN_DEPENDS=	bash>=0:shells/bash \
		wireguard-tools>0:net/wireguard-tools \
		wsireguard-kmod>0:net/wireguard-kmod
```

Also remove duplicated wireguard-tools entry

File: net/pfSense-pkg-WireGuard/Makefile [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+SUB_LIST=	PORTNAME=${PORTNAME}
+
+do-extract:
+	${MKDIR} ${WRKSRC}
+
+do-install:
+	${MKDIR} ${STAGEDIR}/etc/inc/priv
+
+	${MKDIR} ${STAGEDIR}${PREFIX}/pkg
+	${MKDIR} ${STAGEDIR}${PREFIX}/pkg/wireguard
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/Makefile:35]: `${MKDIR}` is `mkdir -p` so when you create `${STAGEDIR}${PREFIX}/pkg/wireguard`, it will automatically create `${STAGEDIR}${PREFIX}/pkg`

File: net/pfSense-pkg-WireGuard/Makefile [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+	${MKDIR} ${STAGEDIR}${PREFIX}/www
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/Makefile:37]: Same here, this line can be safely removed since next one will create this directory

File: net/pfSense-pkg-WireGuard/Makefile [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/pkg/wireguard/wg.inc \
+		${STAGEDIR}${PREFIX}/pkg/wireguard
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/pkg/wireguard/wg_api.inc \
+		${STAGEDIR}${PREFIX}/pkg/wireguard
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/pkg/wireguard/wg_globals.inc \
+		${STAGEDIR}${PREFIX}/pkg/wireguard
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/pkg/wireguard/wg_install.inc \
+		${STAGEDIR}${PREFIX}/pkg/wireguard
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/pkg/wireguard/wg_validate.inc \
+		${STAGEDIR}${PREFIX}/pkg/wireguard
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/Makefile:58]: You can simplify it doing:
```
${INSTALL_DATA} ${FILESDIR}${PREFIX}/pkg/wireguard/*.inc \
	${STAGEDIR}${PREFIX}/pkg/wireguard
```

File: net/pfSense-pkg-WireGuard/Makefile [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+		${STAGEDIR}${PREFIX}/www/shortcuts
+
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/www/wg/status_wireguard.php \
+		${STAGEDIR}${PREFIX}/www/wg
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/www/wg/vpn_wg.php \
+		${STAGEDIR}${PREFIX}/www/wg
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/www/wg/vpn_wg_edit.php \
+		${STAGEDIR}${PREFIX}/www/wg
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/www/wg/vpn_wg_settings.php \
+		${STAGEDIR}${PREFIX}/www/wg
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/Makefile:70]: Same here, can be simplified using wildcard `*.php`

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+require_once('pkg-utils.inc');
+require_once('service-utils.inc');
+require_once('system.inc');
+require_once('util.inc');
+
+// WireGuard includes
+require_once('wg_api.inc');
+require_once('wg_globals.inc');
+require_once('wg_install.inc');
+require_once('wg_validate.inc');
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc:38]: All these files are installed under `/usr/local/pkg/wireguard`, but this directory is not part of PHP include_path variable.  You should prefix them with `wireguard/` because `/usr/local/pkg` is listed on include_path

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc [hunk 10]
```diff
@@ -0,0 +1,416 @@
...
+		// Then attempt to bring up interface using wg-quick(8)
+		exec_wg_quick_action($tunnel, "up", $verbose);
+
+	}
+
+	// Now let's always soft configure the interface
+	wg_syncconf($tunnel);
+
+	// Add the interface to the interface group (i.e. WireGuard)
+	exec("{$wgg['ifconfig']} {$tunnel['name']} group {$wgg['ifgroupentry']['ifname']}" . verbose_pipe($verbose));
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc:59]: It's preferred to use `mwexec()` to execute shell commands

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+		$idx = 0;
+
+		for (;;) {
+			if (!$post['publickeyp'.$idx] &&
+			    !$post['endpoint'.$idx] &&
+			    !$post['allowedips'.$idx] &&
+			    !$post['descr'.$idx] &&
+			    !$post['persistentkeepalive'.$idx]) {
+				break;
+			}
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc:184]: This code would be more clean using a while statement instead of infinite for with a conditional break

```php
		while ($post['publickeyp'.$idx] &&
		    $post['endpoint'.$idx] &&
		    $post['allowedips'.$idx] &&
		    $post['descr'.$idx] &&
		    $post['persistentkeepalive'.$idx]) {
```

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc [hunk 12]
```diff
@@ -0,0 +1,416 @@
...
+				wg_configure_if($tunnel, $conf_hard, $verbose);
+
+			} else {
+
+				// Attempt to destroy the WireGuard tunnel
+				wg_destroy_if($tunnel, $verbose);
+
+			}
+
+
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc:253]: Remove extra blank lines

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+require_once('globals.inc');
+require_once('config.inc');
+require_once('pfsense-utils.inc');
+require_once('system.inc');
+require_once('pkg-utils.inc');
+require_once('service-utils.inc');
+require_once('util.inc');
+
+// WireGuard includes
+require_once('wg_globals.inc');
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:33]: Add `wireguard/` prefix

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+	return htmlspecialchars($output);
+
+}
+
+// Get wg and tools version
+function wg_version() {
+	global $wgg;
+
+	$packages = array();
+	exec("{$wgg['pkg']} info | grep -i {$wgg['pkg_name']}", $packages);
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:74]: Why is pkg binary path stored in `$wgg` array?  It's a regular system binary.  Also it's not needed to use grep, if you want to collect only installed version of a package you know the name you can use `pkg query %v pkg_name`

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+	// Execute wg-quick(8) on the interface using the provided action 
+	exec("{$wgg['wg_quick']} {$action} {$wg_ifname}" . verbose_pipe($verbose));
+
+}
+
+// Generate a pre-shared key
+function genPSK() {
+	global $wgg;
+
+	$psk = exec("{$wgg['wg']} genpsk");
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:152]: It's not considering return code so if command fails this function can return unexpected results

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_globals.inc [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+	$config = parse_config(true);
+
+        // Need to initialize as arrays
+        init_config_arr(array('installedpackages', $wgg['pkg_name'], 'config'));
+        init_config_arr(array('installedpackages', $wgg['pkg_name'], 'tunnel'));
+
+        // Reference assignment for reading and writing back to the configure through $wgg
+        $wgg['config'] = &$config['installedpackages'][$wgg['pkg_name']]['config'][0];
+        $wgg['tunnels'] = &$config['installedpackages'][$wgg['pkg_name']]['tunnel'];
+
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_globals.inc:66]: This function is almost entirely indented using spaces instead of TABs

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_globals.inc [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+        'conf_path'             => '/usr/local/etc/wireguard',
+        'conf_paths_to_clean'   => array('/usr/local/etc/wireguard', '/etc/wireguard', '/etc/wg'),
+        'xml_conf_tags'         => 'wireguard',
+        'if_prefix'             => 'tun_wg',
+        'if_group'              => 'WireGuard',
+        'ifgroupentry'          => array('ifname' => 'WireGuard', 'descr' => 'Dynamic Group for WireGuard', 'members' => null),
+        'default_mtu'           => 1420,
+        'default_port'          => 51820,
+        'max_port'              => 65535,
+        'max_tunnels'           => 32768
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_globals.inc:47]: This block is indented using spaces

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_install.inc [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// pfSense includes
+require_once('config.inc');
+require_once('globals.inc');
+
+// WireGuard includes
+require_once('wg_globals.inc');
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_install.inc:27]: Add `wireguard/` prefix

File: net/pfSense-pkg-WireGuard/files/usr/local/www/wg/status_wireguard.php [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+$shortcut_section = "wireguard";
+$pgtitle = array(gettext("Status"), "WireGuard");
+
+// pfSense includes
+require_once("guiconfig.inc");
+require_once("functions.inc");
+require_once("shortcuts.inc");
+
+// WireGuard includes
+require_once("/usr/local/pkg/wireguard/wg.inc");
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/www/wg/status_wireguard.php:39]: `/usr/local/pkg` can be removed

File: net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg.php [hunk 20]
```diff
@@ -0,0 +1,247 @@
...
+##|*DESCR=Allow access to the 'VPN: WireGuard' page.
+##|*MATCH=vpn_wg.php*
+##|-PRIV
+
+// pfSense includes
+require_once("guiconfig.inc");
+require_once("functions.inc");
+
+// WireGuard includes
+require_once("/usr/local/pkg/wireguard/wg.inc");
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg.php:35]: `/usr/local/pkg` can be removed

File: net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_settings.php [hunk 21]
```diff
@@ -0,0 +1,163 @@
+<?php
+/*
+ * vpn_wg.php
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_settings.php:3]: Wrong filename

File: net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_settings.php [hunk 22]
```diff
@@ -0,0 +1,163 @@
...
+##|*DESCR=WireGuard Settings.
+##|*MATCH=vpn_wg_settings.php*
+##|-PRIV
+
+// pfSense includes
+require_once("guiconfig.inc");
+require_once("functions.inc");
+
+// WireGuard includes
+require_once("/usr/local/pkg/wireguard/wg.inc");
```
Comment (rbgarga) [inline net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_settings.php:35]: `/usr/local/pkg` can be removed

**Review by rbgarga — CHANGES_REQUESTED** [review]

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc (hunk 14 above)
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:74]: > Always use full paths when calling an executable (e.g. /usr/bin/grep NOT grep)

Ref : https://docs.netgate.com/pfsense/en/latest/references/developer-style-guide.html#developer-rules

Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:74]: Either way, I'm now using pkg query and was able to produce something a bit cleaner for everyone

image

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc (hunk 15 above)
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:158]: Fair enough, though I've tested by malforming this many different ways, and it looks like exec() returns null if the retval <> 0, so this might not be necessary to perform any additional checks.

**Review by theonemcdonald — COMMENTED** [review]
I have addressed the review points by @rbgarga ! Thanks for the review, always a pleasure cleaning things up and improving.

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc (hunk 11 above)
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg.inc]: I actually copy-pasted this in my revision (my bad) and of course it broke haha, I took some time to refactor this by setting the input field names to `<input name="endpoint[]"` in vpn_wg_edit.php. This removes the need to guess/check looping and we can use a for loop instead.

```
	if (!empty($post['endpoint'])) {

		// Peers section
		$pconfig['peers'] = array();

		$pconfig['peers']['wgpeer'] = array();

		for ($idx = 0; $idx <= count($post['endpoint']); $idx++) {

			$peer = array();
			$peer['publickey'] = $post['publickeyp'][$idx];
			$peer['endpoint'] = $post['endpoint'][$idx];
			$peer['allowedips'] = $post['allowedips'][$idx];
			$peer['descr'] = $post['descp'][$idx];
			$peer['persistentkeepalive'] = $post['persistentkeepalive'][$idx];
			$peer['presharedkey'] = $post['presharedkey'][$idx];
			$peer['port'] = $post['port'][$idx];
			$peer['peerwgaddr'] = $post['peerwgaddr'][$idx];

			$pconfig['peers']['wgpeer'][] = $peer;

		}

	} else {
		unset($pconfig['peers']);
	}
```

**Review by rbgarga — APPROVED** [review]
After @jim-p review I will import it to the tree and make minor adjustments on ports related stuff if needed

File: net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_edit.php [hunk 23]
```diff
@@ -0,0 +1,687 @@
...
+	<button type="submit" id="saveform" name="saveform" class="btn btn-sm btn-primary" value="save" title="<?=gettext('Save tunnel')?>">
+		<i class="fa fa-save icon-embed-btn"></i>
+		<?=gettext("Save")?>
+	</button>
+</nav>
+
+<?php $jpconfig = json_encode($pconfig, JSON_HEX_APOS); ?>
+<?php $jwg_config = json_encode($wgg['config'], JSON_HEX_APOS); ?>
+
+<?php $genkeywarning = gettext("Are you sure you want to overwrite keys?"); ?>
```
Comment (jim-p) [inline net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_edit.php:461]: Please reword this to avoid addressing the user directly. It could be `Overwrite keys?`, `Click <button name> to overwrite keys`, or similar.

File: net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_edit.php [hunk 24]
```diff
@@ -0,0 +1,687 @@
...
+		document.execCommand("copy");
+	});
+
+	attachhandlers()
+
+	function attachhandlers() {
+		// Delete a peer
+		$('[id^=killpeer_]').click(function () {
+			var row = this.id.slice('killpeer_'.length)
+			if (confirm('Are you sure you want to delete peer ' + row + '?')) {
```
Comment (jim-p) [inline net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_edit.php:579]: Please reword this to avoid addressing the user directly ("you").

**Review by jim-p — CHANGES_REQUESTED** [review]

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard.xml [hunk 25]
```diff
@@ -0,0 +1,55 @@
...
+	<version>%%PKGVERSION%%</version>
+	<title>VPN/WireGuard</title>
+	<include_file>/usr/local/pkg/wireguard/wg.inc</include_file>
+	<menu>
+		<name>WireGuard</name>
+		<section>VPN</section>
+		<url>/wg/vpn_wg.php</url>
+	</menu>
+	<menu>
+		<name>WireGuard Status</name>
```
[the remark this answers is not in the document]
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard.xml:40]: This one bit for for a while, if two menu items have the same name, only the first one is shown in the UI.

[the remark this answers is not in the document]
Comment (jim-p) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard.xml:40]: OK, this one can be skipped then.

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc [hunk 26]
```diff
@@ -0,0 +1,236 @@
...
+
+	}
+
+	// Pass the stripped configuration to wg(8)
+	$syncconf_command ="{$wgg['wg']} syncconf {$wg_ifname} <({$wgg['wg_quick']} strip {$wg_ifname})";
+
+	$syncconf_command = escapeshellarg($syncconf_command);
+
+	// We need <() process substitution, so use bash...might as well use bash everywhere?
+	exec("{$wgg['bash']} -c {$syncconf_command}");
```
[the remark this answers is not in the document]
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:130]: Agreed. This is high on the list to rework.

File: net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_edit.php (hunk 24 above)
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/www/wg/vpn_wg_edit.php:579]: This is no longer relevant.

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc [hunk 27]
```diff
@@ -0,0 +1,236 @@
...
+
+}
+
+// Get various package infos and return an array
+function wg_pkg_info() {
+	global $wgg;
+
+	$packages = implode(" ", $wgg['pkg_names']);
+
+	exec("{$wgg['pkg']} query '%n\t%v\t%c' {$packages}", $output);
```
[the remark this answers is not in the document]
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:72]: Corrected. I'm now walking the array and escaping each element.

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_install.inc [hunk 28]
```diff
@@ -0,0 +1,128 @@
...
+
+	wg_ifgroup_deinstall();
+
+	init_config_arr(array('ifgroups', 'ifgroupentry'));
+
+	$a_ifgroups = &$config['ifgroups']['ifgroupentry'];
+
+	$a_ifgroups[] = $wgg['ifgroupentry'];
+
+	write_config("[WireGuard] Interface group ({$wgg['ifgroupentry']['ifname']}) (re)installed.");
```
[the remark this answers is not in the document]
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_install.inc:98]: Will leave it and do some more testing as to whether this can be safely discarded

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_install.inc [hunk 29]
```diff
@@ -0,0 +1,128 @@
...
+
+		if ($ifg['ifname'] == $wgg['ifgroupentry']['ifname']) {
+
+			unset($a_ifgroups[$index]);
+
+		}
+		
+	}
+
+	write_config("[WireGuard] Interface group ({$wgg['ifgroupentry']['ifname']}) de-installed.");
```
[the remark this answers is not in the document]
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_install.inc:122]: Same here

**Review by theonemcdonald — COMMENTED** [review]
Addressed @jim-p points.

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc [hunk 30]
```diff
@@ -0,0 +1,236 @@
...
+	// Looks like we have a tunnel structure
+	if (is_array($tunnel) && isset($tunnel['name'])) {
+
+		// Transform to an interface name
+		$wg_ifname = $tunnel['name'];
+
+	}
+
+	// Pass the stripped configuration to wg(8)
+	$syncconf_command ="{$wgg['wg']} syncconf {$wg_ifname} <({$wgg['wg_quick']} strip {$wg_ifname})";
```
[the remark this answers is not in the document]
Comment (jim-p) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:136]: This one still needs some work with escaping the parameters on commands being passed through here. As it is, the "outer" command is escaped OK but the parameters on the "inner" command could still be a problem. It needs escaping in both places.

[the remark this answers is not in the document]
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:140]: Whoops, fixed now

[the remark this answers is not in the document]
Comment (jim-p) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc:140]: Looks good now, thanks!

**Comment by jim-p** [comment]
What happened in https://github.com/pfsense/FreeBSD-ports/pull/1063/commits/c43cfafab2a2f203abe70a0a55c6095d932ff049 ? There were a ton of what appear to be whitespace only changes that the github UI isn't interpreting, just showing lines removed and added back that don't appear to change.

**Comment by theonemcdonald** [comment]
> What happened in [c43cfaf](https://github.com/pfsense/FreeBSD-ports/commit/c43cfafab2a2f203abe70a0a55c6095d932ff049) ? There were a ton of what appear to be whitespace only changes that the github UI isn't interpreting, just showing lines removed and added back that don't appear to change.

Hmm not sure what's up with the rendering.

There was a pretty significant bug that was brought to my attention. Even though we can define a subnet mask range on IP/Subnet field via addMask(), this mask range is cloned to additional input lines. This meant that a user couldn't allow both `0.0.0.0/0` and `::/0` at the same time. So I capped the subnet range on the entries to 128 to 1 and then added two checkboxes (one for IPv4 and one for IPv6) to handle the entire address spaces (https://www.youtube.com/watch?v=S2wc9UIVjbA).

**Comment by jim-p** [comment]
I was able to view the bug fix part by telling it to hide whitespace changes, but it's still odd that it showed so many lines changing without any discernible change in the content.

File: net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard.xml (hunk 25 above)
[the remark this answers is not in the document]
Comment (theonemcdonald) [reply net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard.xml]: This can be worked around by Naming one of the menu entries `<name>WireGuard </name>` (with a space)

**Comment by PSSGCSim** [comment]
c43cfafab2a2f203abe70a0a55c6095d932ff049 looks like change from CRLF to LF.

Commit: c43cfafab2a2f203abe70a0a55c6095d932ff049
```
$ file net/pfSense-pkg-WireGuard/Makefile
net/pfSense-pkg-WireGuard/Makefile: BSD makefile script, ASCII text
```

Commit: 5bf28b01e40cd9a99d77f7aa74c79f39bb247e89 (the parent)
```
$ file net/pfSense-pkg-WireGuard/Makefile
net/pfSense-pkg-WireGuard/Makefile: BSD makefile script, ASCII text, with CRLF line terminators
```

**Review by jim-p — APPROVED** [review]
Should be OK enough to merge and test

**Comment by rbgarga** [comment]
I've manually merged it after squash all commits.  I also made some changes after import on separate commits.  Thanks!!
